diff --git a/Formula/c-ares.rb b/Formula/c-ares.rb
index b4f560aed5..b95a4f94f1 100644
--- a/Formula/c-ares.rb
+++ b/Formula/c-ares.rb
@@ -24,7 +24,7 @@ class CAres < Formula
 
   def install
     mkdir "build" do
-      system "cmake", "..", "-GNinja", *std_cmake_args
+      system "cmake", "-DCARES_STATIC=ON", "..", "-GNinja", *std_cmake_args
       system "ninja"
       system "ninja", "install"
     end
