diff --git a/Formula/curl.rb b/Formula/curl.rb
index 82f624d6559..2ab4bfa29d7 100644
--- a/Formula/curl.rb
+++ b/Formula/curl.rb
@@ -42,23 +42,30 @@ class Curl < Formula
   uses_from_macos "zlib"
 
   def install
+    ENV.append_to_cflags "-mmacosx-version-min=10.13"
+
     system "./buildconf" if build.head?
 
     args = %W[
+      --enable-static
+      --disable-shared
       --disable-debug
       --disable-dependency-tracking
       --disable-silent-rules
       --prefix=#{prefix}
-      --with-ssl=#{Formula["openssl@1.1"].opt_prefix}
-      --without-ca-bundle
-      --without-ca-path
+      --without-ssl
       --with-ca-fallback
       --with-secure-transport
-      --with-default-ssl-backend=openssl
-      --with-libidn2
-      --with-librtmp
-      --with-libssh2
       --without-libpsl
+      --disable-ldap
+      --disable-ldaps
+      --with-nghttp2=/usr/local/opt/nghttp2
+      --with-zlib=/usr/local/opt/zlib
+      --enable-ares
+      --without-zstd
+      --without-librtmp
+      --without-libidn2
+      --without-libssh2
     ]
 
     args << if OS.mac?
