diff --git a/Formula/curl.rb b/Formula/curl.rb
index 3df151b466..66ebb7ccc5 100644
--- a/Formula/curl.rb
+++ b/Formula/curl.rb
@@ -39,6 +39,8 @@ class Curl < Formula
     system "./buildconf" if build.head?

     args = %W[
+      --enable-static
+      --disable-shared
       --disable-debug
       --disable-dependency-tracking
       --disable-silent-rules
@@ -46,6 +48,16 @@ class Curl < Formula
       --with-secure-transport
       --without-ca-bundle
       --without-ca-path
+      --without-ssl
+      --disable-ldap
+      --disable-ldaps
+      --with-nghttp2=/usr/local/opt/nghttp2
+      --with-zlib=/usr/local/opt/zlib
+      --enable-ares
+      --without-zstd
+      --without-librtmp
+      --without-libidn2
+      --without-libpsl
     ]

     system "./configure", *args
