diff --git a/Formula/nghttp2.rb b/Formula/nghttp2.rb
index c298c365d05..6c2e521edd5 100644
--- a/Formula/nghttp2.rb
+++ b/Formula/nghttp2.rb
@@ -41,6 +41,8 @@ class Nghttp2 < Formula
   end
 
   def install
+    ENV.append_to_cflags "-mmacosx-version-min=10.13"
+
     # fix for clang not following C++14 behaviour
     # https://github.com/macports/macports-ports/commit/54d83cca9fc0f2ed6d3f873282b6dd3198635891
     inreplace "src/shrpx_client_handler.cc", "return dconn;", "return std::move(dconn);"
